retrieval/transform_test.go

```diff
@@ -760,6 +761,53 @@ func TestSampleBuilder(t *testing.T) {
 				},
 			},
 		},
+		// Samples with a NaN value should be dropped.
+		{
+			targets: targetMap{
+				"job1/instance1": &targets.Target{
+					Labels:           promlabels.FromStrings("job", "job1", "instance", "instance1"),
+					DiscoveredLabels: promlabels.FromStrings("__resource_a", "resource2_a"),
+				},
+			},
+			series: seriesMap{
+				1: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1_count"),
+			},
+			metadata: metadataMap{
+				"job1/instance1/metric1": &scrape.MetricMetadata{Type: textparse.MetricTypeSummary, Metric: "metric1_count"},
+			},
+			metricPrefix: "test.googleapis.com",
+			input: []tsdb.RefSample{
+				// Need multiple samples, since the first will always be nil due to reset timestamp handling
+				{Ref: 1, T: 2000, V: 5},
+				{Ref: 1, T: 3000, V: 8},
+				{Ref: 1, T: 4000, V: math.NaN()},
```

Comment:
Let's also test when a new value is added after NaN is recorded as this edge case may happen. For example:
```
				{Ref: 1, T: 2000, V: 5},
				{Ref: 1, T: 3000, V: 8},
				{Ref: 1, T: 4000, V: math.NaN()},
				{Ref: 1, T: 5000, V: 9},
```